src/cr/cube/matrix/cubemeasure.py

```diff
@@ -1030,34 +1028,71 @@ def baseline(self):
         return uncond_row_margin[:, None] / np.sum(uncond_row_margin)
 
     @lazyproperty
-    def columns_margin(self):
-        """1D ndarray of np.float64 weighted N for each matrix column."""
-        return np.sum(self._weighted_counts, axis=0)
+    def column_bases(self):
+        """2D ndarray of np.float64 column-proportion denominator for each cell."""
+        # --- Never add across subvariables. If rows dimension is an array, then the
+        # --- column_bases is the same as the counts.
+        return (
+            self.weighted_counts
+            if self._row_dimension_is_array
+            else np.broadcast_to(
+                np.sum(self._weighted_counts, axis=0), self.weighted_counts.shape
```

Comment:
why sum on the private `_weighted_counts` and the shape is from the public `weighted_counts`?

Reply:
The shape of the public weighted counts is definitely what we want in the output. This code used to be separated out into the row/column margin and then broadcast to be the shape of the counts, but now that it's closer together I agree it's confusing. 

I've changed them use the public `weighted_counts` for both parts in these cases. The math is the same.